Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1083,6 +1089,46 @@ void HackStudioWidget::initialize_menubar(GUI::Menubar& menubar)
     create_help_menubar(menubar);
 }
 
+void HackStudioWidget::update_statusbar()
+{
+    StringBuilder builder;
+    builder.appendff("Line: {}, Column: {}", current_editor().cursor().line() + 1, current_editor().cursor().column());
+
+    if (current_editor().has_selection()) {
+        size_t first_line;
+        size_t last_line;
+        auto selection = current_editor().normalized_selection();
+        if (!selection.is_valid()) {
+            first_line = current_editor().cursor().line();
+            last_line = current_editor().cursor().line();
+        } else {
+            first_line = selection.start().line();
+            last_line = selection.end().line();
+        }
+        if (first_line == last_line || selection.start().column() == 0 || selection.start().column() != 0)
+            last_line += 1;
+
+        builder.appendff("        Selected: {} {}", last_line - first_line, (last_line - first_line) != 1 ? "Lines" : "Line");
+
+        int word_count = 0;
+        bool in_word = false;
+        String selected_text = current_editor().selected_text();
+        for (char c : selected_text) {
```

Comment:
Would it be better to iterate over this with a UTF-8 view and count code points instead?

Reply:
CLion seems to consider all emojis a single character in their selection length info, which seems like reasonable behavior imo

Reply:
The question is whether you count individual codepoints or grapheme clusters.

Reply:
Right, fair point, even CLion doesn't properly handle emojis like 👨‍👩‍👦 it seems. I think just counting codepoints is reasonable behavior for now?

Reply:
I'll be honest, I never heard of code points or grapheme clusters